src/status_im/contact/block.cljs

```diff
@@ -32,8 +34,9 @@
               (chat.models/upsert-chat
                {:chat-id                      chat-id
                 :unviewed-messages-count
-                (- (get-in db [:chats chat-id :unviewed-messages-count])
-                   removed-unseen-count)})
+                (if (pos? unviewed-messages-count)
```

Comment:
is there some cases where it comes out negative? that would be weird?

Reply:
it definitely did (we use an uint so it was throwing an error saving the chat), I could look more into the causes but I might address separately, although I will make sure it's not introduced in this PR